build_tools/benchmarks/reporting/parse_tflite_benchmarks.py

```diff
@@ -0,0 +1,470 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+r"""Script to generate a HTML summary comparing IREE and TFLite latencies and memory usage.
+
+Example usage:
+
+python parse_tflite_benchmarks.py \
+  --iree_version=20220924.276 \
+  --tflite_version=20220924.162 \
+  --platform=server \
+  --input_csv=server_results.csv \
+  --output_path=/tmp/server_summary.html
+
+"""
+
+import argparse
+import pandas as pd
+import pathlib
+import sys
+
+from datetime import date
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent / ".." / ".." / "python"))
+from reporting.common import html_utils
+
+# Supported platforms.
+_PLATFORM_SERVER = "server"
+_PLATFORM_MOBILE = "mobile"
+
+# A map of model name to data type.
+_MODEL_TO_DATA_TYPE = {
+    "deeplabv3": "fp32",
+    "deeplabv3_fp16": "fp16",
+    "mobilebert-baseline-tf2-quant": "int8",
+    "mobilebert_float_384_gpu": "fp32",
+    "mobilebert_float_384_gpu_fp16": "fp16",
+    "mobilenet_v2_1.0_224": "fp32",
+    "mobilenet_v2_1.0_224_fp16": "fp16",
+    "mobilenet_v2_224_1.0_uint8": "uint8",
+    "person_detect": "int8",
+    "resnet_v2_101_1_default_1": "fp32",
+    "resnet_v2_101_1_default_1_fp16": "fp16",
+    "ssd_mobilenet_v2_static_1.0_int8": "int8",
+}
+
+# Column headers.
+_MODEL = "model"
+_DATA_TYPE = "data type"
+_RUNTIME = "runtime"
+_LATENCY = "latency (ms)"
+_TASKSET = "taskset"
+_MEMORY = "vmhwm (KB)"
+_THREADS = "threads"
+_CONFIG = "config"
+_DRIVER = "driver/delegate"
+_TFLITE_CONFIG = "TFLite config"
+_IREE_CONFIG = "IREE config"
+_IREE_LATENCY = "IREE latency (ms)"
+_TFLITE_LATENCY = "TFLite latency (ms)"
+_IREE_MEMORY = "IREE vmhwm (kb)"
+_TFLITE_MEMORY = "TFLite vmhwm (kb)"
+
+_IREE_VS_TFLITE_LATENCY = "IREE vs TFLite latency"
```

Comment:
The text in this column changes depending on whether IREE is faster or slower i.e. 1.2x faster or 2x slower. So it could either be IREE/TFLite latency if IREE is faster or TFLite/IREE latency if TFLite is faster.